docs/src/guides/scalar_index.md

```diff
@@ -0,0 +1,108 @@
+# Building Scalar Index
+
+Similar to many SQL databases, LanceDB supports several types of Scalar indices to accelerate search
+over scalar columns.
+
+- `BTREE`: The most common type is BTREE. This index is inspired by the btree data structure
+  although only the first few layers of the btree are cached in memory.
+  It will perform well on columns with a large number of unique values and few rows per value.
+- `BITMAP`: this index stores a bitmap for each unique value in the column.
+  This index is useful for columns with a finite number of unique values and many rows per value.
+  For example, columns that represent "categories", "labels", or "tags"
+- `LABEL_LIST`: a special index that is used to index list columns whose values have a finite set of possibilities.
+  For example, a column that contains lists of tags (e.g. `["tag1", "tag2", "tag3"]`) can be indexed with a `LABEL_LIST` index.
+
+| Data Type                                     | Filter                           | Index Type   |
+| --------------------------------------------- | -------------------------------- | ------------ |
+| Numeric, String, Temporal                      | `<`, `=`, `>`, `in`, `between`, `is null`                    | `BTREE`      |
+| Boolean, numbers or strings with finite range of values         | `<`, `=`, `>`, `in`, `between`, `is null`   | `BITMAP`     |
+| List of low cardinality of numbers or strings | `array_has_any`, `array_has_all` | `LABEL_LIST` |
+
+=== "Python"
+
+    ```python
+    import lancedb
+    data = [
+      {"my_col": 1, "category": "A", "vector": [1, 2]},
+      {"my_col": 2, "category": "B", "vector": [3, 4]},
```

Comment:
Don't have to fix this in this PR, but I feel like it would be easier to understand the examples if they had data that would make sense in the real world. For example, we could instead make the example data an array of metadata about books:

```suggestion
    books = [
      {"book_id": 1, "publisher": "plenty of books", "tags": ["fantasy", "adventure"]},
      {"book_id": 2, "publisher": "book town", "tags": ["non-fiction"]},
```